File: entry_point/java_entry_point/EntryPoint.java
import java.lang.*;
//comment

public class EntryPoint {
  public static void main( String[] args ) {
    if( args.length != 1 ) {
      System.err.println( "Usage: java -jar EntryPoint.jar <arg1>" );
      System.exit(1);
    }
    System.out.println("args[0]=" + args[0] );
  }
}

File: entry_point/clojure_entry_point/src/entry_point.clj
(ns entry_point
  (:require [clojure.tools.cli :refer [parse-opts]])
  (:gen-class))

(def cli-options
  ;; An option with a required argument
  [
    ["-h" "--help"]
  ]
)

(defn -main [& args]
  (if (not (= 1 (count args)))
    ((println "Usage: entry_point <arg1>")
    ;(println "Usage: entry_point <arg1>")
    (System/exit 1))
  )
  (println (if args "<one or more args passed>" "<noargs passed>"))
    ;(System/exit 2)
  (parse-opts args cli-options)
  (println args)
  (println (count args))
)
